### PR TITLE
emit a profiler settings event with perf_events_paranoid setting

### PR DESCRIPTION
**Pull request by richardstartin**
# What Does This Do

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public void publish() {
       new ProfilerSettingEvent("Checkpoints", String.valueOf(checkpointsEnabled)).commit();
       new ProfilerSettingEvent("Endpoints", String.valueOf(endpointsEnabled)).commit();
       new ProfilerSettingEvent("Auxiliary Profiler", auxiliaryProfiler).commit();
+      new ProfilerSettingEvent("perf_events_paranoid", perfEventsParanoid).commit();
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java:34]: Nit: Can we have it as `Perf Events Level` or something similar in nicer formatting?

File: dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilerSettingsSupport.java [hunk 2]
```diff
@@ -71,8 +77,23 @@ protected ProfilerSettingsSupport() {
...
             ProfilingConfig.PROFILING_AUXILIARY_TYPE,
             ProfilingConfig.PROFILING_AUXILIARY_TYPE_DEFAULT);
+    perfEventsParanoid = readPerfEventsParanoidSetting();
   }
 
   /** To be defined in controller specific way. Eg. one could emit JFR events. */
   public abstract void publish();
+
+  private static String readPerfEventsParanoidSetting() {
+    String value = "unknown";
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilerSettingsSupport.java:87]: This value will not change while the JVM is still running, right?
Might make sense to cache it as the recording settings are dumped each N seconds (60 by default) with the recording.

Comment (richardstartin) [reply dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilerSettingsSupport.java:87]: I thought that's how this was working with the config memento?

File: dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java (hunk 1 above)
Comment (richardstartin) [reply dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java:34]: I did consider that but this is a very well known setting and it's immediately clear what "perf_events_paranoid" means, but "Perf Events Level" needs explanation.

File: dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilerSettingsSupport.java (hunk 2 above)
Comment (jbachorik) [reply dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilerSettingsSupport.java:87]: Aargh, scratch that. I am embarassing myself. You are completely right.

File: dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java (hunk 1 above)
Comment (jbachorik) [reply dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/JfrProfilerSettings.java:34]: Fair enough

**Review by jbachorik — APPROVED** [review]

**Pull request merged**